### ISSUE TITLE
Function instance of `TransferMechanism.termination_measure` is shared among all instances.

### ISSUE THREAD

**Issue by jvesely** [PrincetonUniversity/PsyNeuLink#1479]
the parameter is declared:
```
termination_measure = Parameter(Distance(metric=MAX_ABS_DIFF), modulable=False, stateful=False, loggable=False)
```
since it's a class attribute of the Prameters class the instance of `Distance` is shared among all instances of TransferMechanism.

**Comment by kmantel** [comment]
In this case, the problem isn't that the `Distance` instance is used as a class default for `termination_measure`, it's that there is an instance created as the default argument to `termination_measure` in `__init__`: https://github.com/PrincetonUniversity/PsyNeuLink/blob/6c7fb550fbe2efb25abe59950864409ca1d4590c/psyneulink/core/components/mechanisms/processing/transfermechanism.py#L951

This is created once on import, and then used for every TransferMechanism without that argument specified. In general, the exact object passed in to a constructor will be used as the parameter value. The exception is for the `function` parameter, which would make a copy of the object if it was already used by another `Mechanism` (the function's `owner` is not None).

A quick fix would be to just set the default argument value in `TransferMechanism.__init__` above to None. This would have the default set properly for `termination_measure`, (And in general, they should be None, so the `_user_specified` Parameter flag is set correctly, so this would be a good change regardless) but it would still leave this example with the current behavior:

```
>>> dist = pnl.Distance()
>>> a = pnl.TransferMechanism(termination_measure=dist)
>>> b = pnl.TransferMechanism(termination_measure=dist)
>>> a.defaults.termination_measure is b.defaults.termination_measure
True
```

I think in general though, the `function` behavior should be expanded to other "subfunction" parameters

**Comment by jvesely** [comment]
is the change in init a one-liner that should work, or does it trigger changes elsewhere? Do you have plans to work on it?
The example is OK. It probably won't work with compilation, but the sharing is clearly intended.

**Comment by kmantel** [comment]
The change should work, because the Parameter default will be used when the argument to __init__ is None. Using None should be the correct pattern in general because it will let us know if the user passed in an argument or not. It's been a low priority task to go through and change all those, but it can be bumped up as necessary.

I guess I'm not sure what's the best thing to do with the example. For `function` we definitely wanted protection from instances being shared; I'm not sure if the same pattern should hold for other parameters that happen to take functions.

**Comment by jvesely** [comment]
Fixes in ded82bf
